### PR TITLE
简化 ovn eip 类型

### PR DESCRIPTION
**Pull request by bobz965**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
修复 bfd 路由无法新增多个
修复 vpc 更新路由未触发

### <samp>🤖 Generated by Copilot at 9cae868</samp>

Refactor the code to use new and simplified types for ovn eip and vip scenarios. This improves the readability and consistency of the code and the e2e test, and aligns with the ovn terminology. The main changes are in `pkg/controller`, `pkg/daemon`, `pkg/apis`, `pkg/util`, and `test/e2e`.

### <samp>🤖 Generated by Copilot at 9cae868</samp>

> _We are the masters of the ovn eip_
> _We use the `Lrp`, `Lsp`, and `Vip`_
> _We simplify the types and the names_
> _We crush the bugs and the flames_

### <samp>🤖 Generated by Copilot at 9cae868</samp>

*  Simplify the eip usage types from `eip`, `lrp`, and `node external gw` to `lrp`, `lsp`, and `nat` in the `OvnEipSpec` type definition ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-69f75d53361877cc02ab6f413f306486d29287c6400b92b71c4c74f018f3234aL952-R955))
*  Use the new `Lrp` type instead of the old `LrpUsingEip` type when creating or updating the ovn eip cr for the logical router port in `createDefaultVpcLrpEip` and `handleAddVpcExternal` functions ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-49e9f500bbdd1972f437e33413a4e9761aaa80c62cb2b31fe92b28c8b069333fL203-R203), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-ae8a8b5d44d277a4bb96635ee325fe54b48e8bc10b24a79d4bb8e211b57131ecL815-R815))
*  Use the new `Lsp` type instead of the old `NodeExtGwUsingEip` type when creating, updating, or deleting the bare logical switch port for the eip in `handleAddOvnEip`, `handleUpdateOvnEip`, and `handleDelOvnEip` functions ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL250-R250), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL265-R265), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL305-R305), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL348-R348))
*  Use the new `Lrp` type instead of the old `LrpUsingEip` type when deleting the logical router port for the eip in `handleDelOvnEip` function ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL355-R355))
*  Use the new `Vip` type instead of the old `NatUsingVip` type when getting the internal vip from the virtual ip lister in `handleAddOvnFip`, `handleUpdateOvnFip`, `handleAddOvnDnatRule`, and `handleUpdateOvnDnatRule` functions ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L218-R218), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L357-R357), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1d005a7fb842ce7ae53c4bd4780cf0703a409b350f4f41d482b36158af0f459L207-R207), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1d005a7fb842ce7ae53c4bd4780cf0703a409b350f4f41d482b36158af0f459L313-R313))
*  Use the new `Lsp` type instead of the old `NodeExtGwUsingEip` type when listing or checking the node external ovn eips in `reconcileCustomVpcBfdStaticRoute` and `checkNodeGwNicInNs` functions ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L1193-R1193), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L1199-R1199), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-9ef24f4ecb94bef2f6117b5b04dd931cc4cec65ab035472384bce083898827feL446-R446), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-9ef24f4ecb94bef2f6117b5b04dd931cc4cec65ab035472384bce083898827feL453-R453))
*  Define the new eip usage types `Vip`, `Lrp`, and `Lsp` instead of the old types `NatUsingVip`, `LrpUsingEip`, and `NodeExtGwUsingEip` in the `util` package ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-23fe1930ce750ec27bfa79a8fb8d076245c64df2b62ea6d6367cdd59260b164cL178-R184))
*  Use the new `Vip` type instead of the old `NatUsingVip` type when creating the ovn fip and dnat with the share eip vip in the e2e test ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L448-R454), [link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L558-R558))
*  Use the new `Lsp` type instead of the old `NodeExtGwUsingEip` type when creating the ovn node-ext-gw type eip on each node in the e2e test ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L505-R505))
*  Use the new `Vip` type instead of the old `NatUsingVip` type when creating the ovn fip with the fip eip vip in the e2e test ([link](https://github.com/kubeovn/kube-ovn/pull/3107/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L541-R541))

### CODE REVIEW

File: pkg/util/const.go [hunk 1]
```diff
@@ -175,13 +175,13 @@ const (
 	EcmpRouteType   = "ecmp"
 	NormalRouteType = "normal"
 
-	LrpUsingEip        = "lrp"
+	Vip                = "vip"
+	Lrp                = "lrp"
+	Lsp                = "lrp"
```
Comment (hongzhen-ma) [inline pkg/util/const.go:180]: 这个参数应该是错了吧

Comment (hongzhen-ma) [inline pkg/util/const.go:180]: 这个参数应该是错了吧

Comment (bobz965) [reply pkg/util/const.go:180]: 是的

File: pkg/controller/ovn_fip.go [hunk 2]
```diff
@@ -258,6 +265,12 @@ func (c *Controller) handleAddOvnFip(key string) error {
 		err = fmt.Errorf("ovn eip %s type is not %s, can not use", cachedEip.Name, util.NatUsingEip)
 		return err
 	}
+	if cachedEip.Spec.Type == util.Lsp {
```
Comment (hongzhen-ma) [inline pkg/controller/ovn_fip.go:268]: 这个检查，跟上边这个，不重复吗？

File: pkg/controller/vpc.go [hunk 3]
```diff
@@ -281,9 +276,10 @@ func (c *Controller) handleAddOrUpdateVpc(key string) error {
 		klog.Errorf("failed to get vpc %s static route list, %v", vpc.Name, err)
 		return err
 	}
-
 	rtbs := c.getRouteTablesByVpc(vpc)
 	targetRoutes := vpc.Spec.StaticRoutes
+	klog.Infof("vpc %s spec static routes: %v", key, targetRoutes)
+	klog.Infof("vpc %s exist route: %v", key, existRoute)
```
Comment (hongzhen-ma) [inline pkg/controller/vpc.go:282]: log 能合并吗？

File: pkg/controller/vpc.go [hunk 4]
```diff
@@ -370,6 +366,8 @@ func (c *Controller) handleAddOrUpdateVpc(key string) error {
 		klog.Errorf("failed to diff vpc %s static route, %v", vpc.Name, err)
 		return err
 	}
+	klog.Infof("vpc %s add routes: %v", key, routeNeedAdd)
+	klog.Infof("vpc %s del routes: %v", key, routeNeedDel)
```
Comment (hongzhen-ma) [inline pkg/controller/vpc.go:370]: 最好分开，在实际处理之前加 log

add route log 可以放在下边

File: pkg/controller/vpc.go (hunk 3 above)
Comment (bobz965) [reply pkg/controller/vpc.go:282]: ok

File: pkg/controller/ovn_fip.go (hunk 2 above)
Comment (bobz965) [reply pkg/controller/ovn_fip.go:268]: 重复了

File: pkg/controller/vpc.go (hunk 4 above)
Comment (bobz965) [reply pkg/controller/vpc.go:370]: 好的 我感觉去掉更好

**Review by hongzhen-ma — APPROVED** [review]

**Comment by oilbeater** [comment]
cleanup 大量失败，需要检查一下相关逻辑

**Pull request merged**